front/lib/production_checks/checks/check_data_sources_consistency.ts

```diff
@@ -0,0 +1,71 @@
+import { CheckFunction } from "@app/lib/production_checks/types";
+import { DataSourceModel } from "@app/lib/resources/storage/models/data_source";
+import { DataSourceViewModel } from "@app/lib/resources/storage/models/data_source_view";
+import { Op, Sequelize } from "sequelize";
+
+export const checkDataSourcesConsistency: CheckFunction = async (
```

Comment:
Rather than a production check, can we instead use a transaction in the `DataSourceResource.makeNew()` to create the default associated view?

Also this check might still be relevant for the `connectorId` on data source 🙃.

Reply:
DataSourceResource cannot depend on DataSourceViewResource as we enforce no dependency cycle :-) 
But could be ok to do the opposite move the creation code DataSourceViewResource  instead ?